### PR TITLE
fix: break up multi-flag args

### PR DESCRIPTION
**Pull request by davidvader**
fixes an issue across all flags where go.os.Exec uses the entire flag as a raw string, causing terraform to interpret a multi-flag as a single string argument.

closes: https://github.com/go-vela/community/issues/331

### CODE REVIEW

**Review by wass3r — APPROVED** [review]
thank you 🙇🏼

File: cmd/vela-terraform/apply.go [hunk 1]
```diff
@@ -123,23 +123,20 @@ func (a *Apply) Command(dir string) *exec.Cmd {
 
 	// check if Vars is provided
 	if len(a.Vars) > 0 {
-		var vars string
 		for _, v := range a.Vars {
-			vars += fmt.Sprintf("-var=%s ", v)
+			flag := fmt.Sprintf(`-var=%s`, v)
+			// add flag for Vars from provided command
+			flags = append(flags, flag)
```
Comment (jbrockopp) [inline cmd/vela-terraform/apply.go:129]: Do you think we could condense this into one line?

```suggestion
			// add flag for Vars from provided command
			flags = append(flags, fmt.Sprintf(`-var=%s`, v))
```

File: cmd/vela-terraform/apply.go [hunk 2]
```diff
@@ -123,23 +123,20 @@ func (a *Apply) Command(dir string) *exec.Cmd {
...
 	}
 
 	// check if VarFiles is provided
 	if len(a.VarFiles) > 0 {
-		var files string
 		for _, v := range a.VarFiles {
-			files += fmt.Sprintf("-var-file=%s ", v)
+			flag := fmt.Sprintf(`-var-file=%s`, v)
+			// add flag for VarFiles from provided command
+			flags = append(flags, flag)
```
Comment (jbrockopp) [inline cmd/vela-terraform/apply.go:138]: Do you think we could condense this into one line?

```suggestion
			// add flag for VarFiles from provided command
			flags = append(flags, fmt.Sprintf(`-var-file=%s`, v))
```

**Review by jbrockopp — CHANGES_REQUESTED** [review]

**Review by wass3r — APPROVED** [review]

File: cmd/vela-terraform/destroy.go [hunk 3]
```diff
@@ -115,23 +115,20 @@ func (a *Destroy) Command(dir string) *exec.Cmd {
 
 	// check if Vars is provided
 	if len(a.Vars) > 0 {
-		var vars string
 		for _, v := range a.Vars {
-			vars += fmt.Sprintf("-var=%s ", v)
+			flag := fmt.Sprintf(`-var=%s`, v)
+			// add flag for Vars from provided command
+			flags = append(flags, flag)
```
Comment (jbrockopp) [inline cmd/vela-terraform/destroy.go:121]: Do you think we could condense this into one line?

```suggestion
			// add flag for Vars from provided command
			flags = append(flags, fmt.Sprintf(`-var=%s`, v))
```

File: cmd/vela-terraform/destroy.go [hunk 4]
```diff
@@ -115,23 +115,20 @@ func (a *Destroy) Command(dir string) *exec.Cmd {
...
 	}
 
 	// check if VarFiles is provided
 	if len(a.VarFiles) > 0 {
-		var files string
 		for _, v := range a.VarFiles {
-			files += fmt.Sprintf("-var-file=%s ", v)
+			flag := fmt.Sprintf(`-var-file=%s`, v)
+			// add flag for VarFiles from provided command
+			flags = append(flags, flag)
```
Comment (jbrockopp) [inline cmd/vela-terraform/destroy.go:130]: Do you think we could condense this into one line?

```suggestion
			// add flag for VarFiles from provided command
			flags = append(flags, fmt.Sprintf(`-var-file=%s`, v))
```

File: cmd/vela-terraform/init.go [hunk 5]
```diff
@@ -81,11 +81,10 @@ func (i *Init) Command(dir string) *exec.Cmd {
 
 	// check if BackendConfigs is provided
 	if len(i.InitOptions.BackendConfigs) > 0 {
-		var configs string
 		for _, v := range i.InitOptions.BackendConfigs {
-			configs = fmt.Sprintf(`-backend-config=%s`, v)
+			flag := fmt.Sprintf(`-backend-config=%s`, v)
 			// add flag for BackendConfigs from provided init command
-			flags = append(flags, configs)
+			flags = append(flags, flag)
```
Comment (jbrockopp) [inline cmd/vela-terraform/init.go:87]: Do you think we could condense this into one line?

```go
flags = append(flags, fmt.Sprintf(`-backend-config=%s`, v))
```

File: cmd/vela-terraform/init.go [hunk 6]
```diff
@@ -139,13 +138,11 @@ func (i *Init) Command(dir string) *exec.Cmd {
 
 	// check if PluginDirs is provided
 	if len(i.InitOptions.PluginDirs) > 0 {
-		var configs string
 		for _, v := range i.InitOptions.PluginDirs {
-			configs += fmt.Sprintf("-plugin-dir=%s ", v)
+			flag := fmt.Sprintf(`-plugin-dir=%s`, v)
+			// add flag for PluginDirs from provided init command
+			flags = append(flags, flag)
```
Comment (jbrockopp) [inline cmd/vela-terraform/init.go:144]: Do you think we could condense this into one line?

```suggestion
			// add flag for PluginDirs from provided init command
			flags = append(flags, fmt.Sprintf(`-plugin-dir=%s`, v))
```

File: cmd/vela-terraform/plan.go [hunk 7]
```diff
@@ -131,23 +131,20 @@ func (p *Plan) Command(dir string) *exec.Cmd {
 
 	// check if Vars is provided
 	if len(p.Vars) > 0 {
-		var vars string
 		for _, v := range p.Vars {
-			vars += fmt.Sprintf("-var=%s ", v)
+			flag := fmt.Sprintf(`-var=%s`, v)
+			// add flag for Vars from provided command
+			flags = append(flags, flag)
```
Comment (jbrockopp) [inline cmd/vela-terraform/plan.go:137]: Do you think we could condense this into one line?

```suggestion
			// add flag for Vars from provided command
			flags = append(flags, fmt.Sprintf(`-var=%s`, v))
```

File: cmd/vela-terraform/plan.go [hunk 8]
```diff
@@ -131,23 +131,20 @@ func (p *Plan) Command(dir string) *exec.Cmd {
...
 	}
 
 	// check if VarFiles is provided
 	if len(p.VarFiles) > 0 {
-		var files string
 		for _, v := range p.VarFiles {
-			files += fmt.Sprintf("-var-file=%s ", v)
+			flag := fmt.Sprintf(`-var-file=%s`, v)
+			// add flag for VarFiles from provided command
+			flags = append(flags, flag)
```
Comment (jbrockopp) [inline cmd/vela-terraform/plan.go:146]: Do you think we could condense this into one line?

```suggestion
			// add flag for VarFiles from provided command
			flags = append(flags, fmt.Sprintf(`-var-file=%s`, v))
```

File: cmd/vela-terraform/validate.go [hunk 9]
```diff
@@ -50,23 +50,20 @@ func (v *Validation) Command(dir string) *exec.Cmd {
 
 	// check if Vars is provided
 	if len(v.Vars) > 0 {
-		var vars string
 		for _, v := range v.Vars {
-			vars += fmt.Sprintf("-var=%s ", v)
+			flag := fmt.Sprintf(`-var=%s`, v)
+			// add flag for Vars from provided command
+			flags = append(flags, flag)
```
Comment (jbrockopp) [inline cmd/vela-terraform/validate.go:56]: Do you think we could condense this into one line?

```suggestion
			// add flag for Vars from provided command
			flags = append(flags, fmt.Sprintf(`-var=%s`, v))
```

File: cmd/vela-terraform/validate.go [hunk 10]
```diff
@@ -50,23 +50,20 @@ func (v *Validation) Command(dir string) *exec.Cmd {
...
 	}
 
 	// check if VarFiles is provided
 	if len(v.VarFiles) > 0 {
-		var files string
 		for _, v := range v.VarFiles {
-			files += fmt.Sprintf("-var-file=%s ", v)
+			flag := fmt.Sprintf(`-var-file=%s`, v)
+			// add flag for VarFiles from provided command
+			flags = append(flags, flag)
```
Comment (jbrockopp) [inline cmd/vela-terraform/validate.go:65]: Do you think we could condense this into one line?

```suggestion
			// add flag for VarFiles from provided command
			flags = append(flags, fmt.Sprintf(`-var-file=%s`, v))
```

**Review by jbrockopp — CHANGES_REQUESTED** [review]
Apologies I should have mentioned these in my first review 😞

File: cmd/vela-terraform/init.go (hunk 5 above)
Comment (davidvader) [reply cmd/vela-terraform/init.go]: done 👍

**Comment by davidvader** [comment]
@jbrockopp condensed those flags 👍

**Review by jbrockopp — APPROVED** [review]
LGTM

**Review by wass3r — APPROVED** [review]

**Pull request merged**